vulture_os/toolkit/api_parser/beyondtrust_reportings/beyondtrust_reportings.py

```diff
@@ -0,0 +1,325 @@
+                headers.update({'Authorization': f"Bearer {self.api_token}"})
+                self.session.headers.update(headers)
+        except Exception as err:
+            raise BeyondtrustReportingsAPIError(f"Error on _connect(): {err}")
+
+    def _execute_query(self, url, query={}, timeout=20):
+        msg = f"URL : {url} Query : {str(query)}"
+        logger.debug(f"[{__parser__}] Request API : {msg}", extra={'frontend': str(self.frontend)})
+
+        retry = 0
+        while retry < 2 and not self.evt_stop.is_set():
+            if self.session is None:
+                self._connect()
+            response = self.session.get(
+                url,
+                params=query,
+                timeout=timeout,
+                proxies=self.proxies,
+                verify=self.api_parser_custom_certificate or self.api_parser_verify_ssl
+            )
+            # Handle rate-limiting
+            if response.status_code == 429:
+                logger.warning(f"[{__parser__}]:execute: API Rate limit exceeded, waiting 10 seconds...",
+                            extra={'frontend': str(self.frontend)})
+                self.evt_stop.wait(10.0)
+                retry += 1
+                continue
+            # Handle token expiration
+            elif response.status_code == 401:
+                logger.warning(f"[{__parser__}]:execute: 401 received, will try to re-authenticate...",
+                            extra={'frontend': str(self.frontend)})
+                self.session = None
+                self.beyondtrust_reportings_api_token = None
+                self.evt_stop.wait(5.0)
+                retry += 1
+                continue
+            elif response.status_code == 200:
+                return response
+
+        raise BeyondtrustReportingsAPIError(
+            f"Error at Beyondtrust PRA API Call URL: {url} Code: {response.status_code} Content: {response.content}")
+
+    def test(self):
+        # Get the last 12 hours
+        since = timezone.now() - timedelta(hours=12)
+        err = ""
+        for requested_type in self.allowed_reporting_types.keys():
+            try:
+                logs, _ = self.get_logs("reporting", requested_type, since)
+
+                return {
+                    "status": True,
+                    "data": logs
+                }
+            except Exception as e:
+                err = e
+                continue
+
+        logger.exception(f"{[__parser__]}:test: {str(err)}", extra={'frontend': str(self.frontend)})
+        return {
+            "status": False,
+            "error": str(err)
+        }
+
+    @staticmethod
+    def format_team_logs(logs):
+        formated_logs = []
+        logs = logs['team_activity_list'].get('team_activity', [])
+        if not isinstance(logs, list):
+            logs = [logs]
+
+        for log in logs:
+            events = log['events']['event']
+            if not isinstance(events, list):
+                events = [events]
```

Comment:
In formatting functions (post receiving logs) you should maybe avoid exceptions caused by missing keys: it should be better to log the error instead and keep processing the other types of logs